cli/CMakeLists.txt

```diff
@@ -30,6 +30,12 @@ if(LIBNUMA_FOUND)
   set(QUICKSTEP_HAVE_LIBNUMA TRUE)
 endif()
 
+if (BUILD_SHARED_LIBS)
+  set(GFLAGS_LIB_NAME gflags_nothreads-shared)
+else()
+  set(GFLAGS_LIB_NAME gflags_nothreads-static)
+endif()
```

Comment:
Yes, as I noted in the comment below, we can definitely do that. But 
- CMake is *not* smart enough to figure out the double-linking on its own.
- Specifically for gflags, there's no common target name that we can use to link whichever static/shared automatically so that CMake can resolve the library name on its own. This works for all other libraries, third_party and our own. Just not for gflags.
- The NOT TARGET test will have to be added everywhere that a third_party library is being linked, in all the CMakeLists. That's a *lot* of changes for very limited benefit.